ethcore/src/blockchain/block_info.rs

```diff
@@ -40,12 +42,55 @@ pub enum BlockLocation {
 	/// It's part of the fork which should become canon chain,
 	/// because it's total difficulty is higher than current
 	/// canon chain difficulty.
-	BranchBecomingCanonChain {
-		/// Hash of the newest common ancestor with old canon chain.
-		ancestor: H256,
-		/// Hashes of the blocks between ancestor and this block.
-		enacted: Vec<H256>,
-		/// Hashes of the blocks which were invalidated.
-		retracted: Vec<H256>,
+	BranchBecomingCanonChain(BranchBecomingCanonChainData),
```

Comment:
@gavofyork 
yes this is possible, but it will take some time and will be targeted by future prs
the idea is to generate code that will write/read structs directly to/from nanomsg buffers and to achieve zero-copy messaging
and this is kind of temporal solution